commit a4264b3f4049ae7aeeb0017f8158119e22fa354f
Author: Jesper Juhl <jj@chaosbits.net>
Date:   Sun Dec 12 23:18:15 2010 +0100

    UDF: Close small mem leak in udf_find_entry()
    
    Hi,
    
    There's a small memory leak in fs/udf/namei.c::udf_find_entry().
    
    We dynamically allocate memory for 'fname' with kmalloc() and in most
    situations we free it before we leave the function, but there is one
    situation where we do not (but should). This patch closes the leak by
    jumping to the 'out_ok' label which does the correct cleanup rather than
    doing half the cleanup and returning directly.
    
    Signed-off-by: Jesper Juhl <jj@chaosbits.net>
    Signed-off-by: Jan Kara <jack@suse.cz>

--- a/fs/udf/namei.c
+++ b/fs/udf/namei.c
@@ -227,10 +227,8 @@ static struct fileIdentDesc *udf_find_en
 		}
 
 		if ((cfi->fileCharacteristics & FID_FILE_CHAR_PARENT) &&
-		    isdotdot) {
-			brelse(epos.bh);
-			return fi;
-		}
+		    isdotdot)
+			goto out_ok;
 
 		if (!lfi)
 			continue;
